torchtext/datasets/ag_news.py

```diff
@@ -36,5 +36,5 @@ def AG_NEWS(root, split):
                              path=os.path.join(root, split + ".csv"),
                              hash_value=MD5[split],
                              hash_type='md5')
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+
+    return CSVParserIterDataPipe([(path, open(path, 'rb'))]).map(lambda t: (int(t[1]), ' '.join(t[2:])))
```

Comment:
You can use torch.utils.data.datapipes.iter.LoadFilesFromDisk
` dp =  LoadFilesFromDisk([path]).parse_csv_files().map(lambda t: (int(t[1]), ' '.join(t[2:])))`